### PR TITLE
Heroku deployment

### PR DESCRIPTION
**Pull request by novaraptur**
# Description

Deployed API to heroku; removed node_modules from git.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Refactor

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Console Dev Tools
- [ ] Cypress

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,17 +5,18 @@
...
   "dependencies": {
     "cors": "^2.8.5",
     "dotenv": "^10.0.0",
     "express": "^4.17.1",
-    "pg": "^8.6.0"
+    "pg": "^8.6.0",
+    "postgresql": "^0.0.1"
   },
-  "devDependencies": {
+  "optionalDependencies": {
```
Comment (shawnmcmahon) [inline package.json:19]: Note to self for future projects: putting Nodemon in devDependecies will cause issues when trying to deploy an API to Heroku.

File: config.js [hunk 2]
```diff
@@ -3,12 +3,16 @@ require('dotenv').config()
 
 
 const isProduction = process.env.NODE_ENV === 'production'
-const connectionString = `postgresql://${process.env.DB_USER}:${process.env.DB_PASSWORD}@${process.env.DB_HOST}:${process.env.DB_PORT}/${process.env.DB_DATABASE}`
+//const connectionString = `postgres://xvkpuofcnndyjr:c3dc401551f597335bd3115c401af0f9c8bbcd8196227e30d40c34ec9fcebead@ec2-52-72-125-94.compute-1.amazonaws.com:5432/d5mfdsej9ljpfe`
```
Comment (hoomberto) [inline config.js:6]: Interesting that these both work, but I think having the connectionString rely on our process.env will make it easier developing

File: config.js [hunk 3]
```diff
@@ -8,7 +8,10 @@ const connectionString = `postgres://xvkpuofcnndyjr:c3dc401551f597335bd3115c401a
 
 const pool = new Pool({
   connectionString: isProduction ? process.env.DATABASE_URL : connectionString,
-  ssl: isProduction,
+  ssl: {
```
Comment (shawnmcmahon) [inline config.js:12]: This was the missing piece that was keeping us from deploying to Heroku. This is important to know for future projects.

File: config.js [hunk 4]
```diff
@@ -3,12 +3,16 @@ require('dotenv').config()
...
+//const connectionString = `postgres://xvkpuofcnndyjr:c3dc401551f597335bd3115c401af0f9c8bbcd8196227e30d40c34ec9fcebead@ec2-52-72-125-94.compute-1.amazonaws.com:5432/d5mfdsej9ljpfe`
+const connectionString = `postgresql://${process.env.DB_USER}:${process.env.DB_PASSWORD}@${process.env.DB_HOST}:${process.env.DB_PORT}/${process.env.DB_DATABASE}`;
 
 
 const pool = new Pool({
   connectionString: isProduction ? process.env.DATABASE_URL : connectionString,
-  ssl: isProduction,
+  ssl: {
+    sslmode: 'require',
+    rejectUnauthorized: false,
```
Comment (hoomberto) [inline config.js:14]: Crazy that this was all we needed to get it working! I wonder if we'll have to consider implementing an API key (perhaps via JSON web tokens? https://jwt.io/introduction/)

File: index.js [hunk 5]
```diff
@@ -22,8 +22,10 @@ app.get('/', (request, response) => {
 app.get('/questions', (request, response) => {
   let questions;
   pool.query('SELECT * FROM questions', (err, res) => {
+    console.log(res);
     if (err) {
       console.log(err)
+      return err;
```
Comment (hoomberto) [inline index.js:28]: Great catch adding returns here!

**Review by hoomberto — APPROVED** [review]

**Pull request merged**